src/qibolab/dummy/platform.py

```diff
@@ -1,83 +1,36 @@
 import pathlib
 
-from qibolab.components import (
-    AcquireChannel,
-    AcquisitionConfig,
-    DcChannel,
-    DcConfig,
-    IqChannel,
-    IqConfig,
-    OscillatorConfig,
-)
+from qibolab.components import AcquireChannel, DcChannel, IqChannel
 from qibolab.instruments.dummy import DummyInstrument, DummyLocalOscillator
-from qibolab.kernels import Kernels
 from qibolab.platform import Platform
-from qibolab.serialize import (
-    load_instrument_settings,
-    load_qubits,
-    load_runcard,
-    load_settings,
-)
 
 FOLDER = pathlib.Path(__file__).parent
 
 
 def create_dummy():
     """Create a dummy platform using the dummy instrument."""
+    # register the instruments
     instrument = DummyInstrument("dummy", "0.0.0.0")
+    twpa_pump = DummyLocalOscillator("twpa_pump", "0.0.0.0")
 
-    twpa_pump_name = "twpa_pump"
-    twpa_pump = DummyLocalOscillator(twpa_pump_name, "0.0.0.0")
+    platform = Platform.load(path=FOLDER, instruments=[instrument, twpa_pump])
```

Comment:
I am working on updating the qw11qD platform (https://github.com/qiboteam/qibolab_platforms_qrc/pull/153) in this layout in order to test on hardware, however I am not sure if I should first initialize the platform or the instruments. In particular:
1. Platform needs the instruments, as evident from this line.
2. Instruments need the "instrument specific" channels (in this case `QmChannel`), which need the logical channels, which are hanging from the `Qubit`, which is now only available after `Platform` creation.

For the dummy you can initialize the platform first because there are no "instrument specific" channels. However, for real configurations this are crucial since they contain the information about the instrument port that the logical channels are connected.

Obviously, an easy work-around is to (ab)use Python's ability to modify attributes after object creation and either attach instruments to the platform after creation (to solve 1) or attach channels to instruments (to solve 2). However, I am not sure if this is optimal in terms of design. It is not necessarily bad, since we are already doing it to attach channels to `Qubit`s.

Another work-around is to create logical channels first, without attaching them to `Qubit`s, use them to create the instrument channels, then the instruments, then the `Platform` and finally attach logical channels to `Qubit`s. This will make platform definitions slightly longer.

@alecandido do you have any preference or different suggestion?

Reply:
This is the type of hint I need to finalize the interface, thanks @stavros11. It's clear that dummy is not representative enough.

I'd suggest you to examine the state in #983, where this problem should already be solved: `Qubit` do not require any longer the information in `parameters.json`. Having to join the two things was already pretty much annoying on its own that I removed it (now the natives are looked up lazily, only when it's needed, and not during platform creation).

Furthermore, `Platform.load()` is just a convenience wrapper, but you should be able to instantiate the `Platform` without. In which case, you can load the `parameters.json` (just with `Paremeters.model_validate(json.loads(path / to / "parameters.json"))` manually).
I'm not sure it is strictly needed, and it should not be for this specific problem after #983, and that's why there is not a more convenient interface (yet).